user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
Actually - [`AuthenticationFailed`](http://tomchristie.github.io/rest-framework-2-docs/api-guide/exceptions#authenticationfailed) is closer to what I meant, though still usually becomes a `401`.